### PR TITLE
blog/content/post/2022-01-17-kgpe_D16_status.md: add KGPE-D16 status …

### PR DESCRIPTION
**Pull request by miczyg1**
…blog post

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

### CODE REVIEW

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+3mdeb started the code refactoring and improving efforts in September 2021. But
+the scope was not only to bring the code quality to coreboot's current
+standards, no. We (3mdeb and the sponsors) aimed to create even more secure and
+libre platform than it was before (back then in main coreboot branch). Besides
+the coreboot requirements 3mdeb has enabled and validated TPM 2.0 and vboot on
+the platform using 8 MB and 16 MB SPI flashes (much larger than 2 MB default
+ones). 3 months later the finalized firmware has been released under
+[Dasharo](https://dasharo.com/) brand with multi-flavored pre-built binaries
+ready to use by end users. Now the platform benefits from automated tests,
```
Comment (pietrushnic) [inline blog/content/post/2022-01-17-kgpe_D16_status.md:112]: and transparen validation, when test results would be published

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Besides coreboot improvements, the project charter also included upstreaming of
+the [flashrom patches](https://review.coreboot.org/c/flashrom/+/59713) for
+write protection and OTP memory management. These were especially intended for
+use with KGPE-D16 platform and the non-standard flashes to form a Static Root
+of Trust with immutable coreboot's bootblock and provide best security possible
+on this platform.
+
+The refreshed code for ASUS KGPE-D16 may be found on [Dasharo GitHub](https://github.com/Dasharo/coreboot/tree/asus_kgpe-d16/develop)
+and will be upstreamed to the main coreboot branch soon.
```
Comment (pietrushnic) [inline blog/content/post/2022-01-17-kgpe_D16_status.md:127]: @miczyg1 please mention Dasharo matrix where we are availabe for discussing and supporting the project.

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+Be sure to attend this years' FOSDEM'22 where I will be talking about current
+[status of AMD platforms](https://fosdem.org/2022/schedule/event/osf_on_amd_3rd/)
+(including KGPE-D16) in open source firmware in the Open Source Firmware, BMC
+and bootloader devroom.
+
+Also feel invited for the FOSDEM'22 after-party organized right after the
+devroom track where we will be discussing various aspects of related to
+firmware and bootloaders. 3mdeb is also organizing a vPub on the 17th of
+February, don't miss this occasion to talk with us, share your ideas, thoughts
+and projects.
```
Comment (pietrushnic) [inline blog/content/post/2022-01-17-kgpe_D16_status.md:146]: Please add link to vPub website.

**Review by pietrushnic — CHANGES_REQUESTED** [review]

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+https://review.coreboot.org/q/topic:raptor-asus-kgpe-d16 
+The same was done for OpenBMC
+https://www.raptorengineering.com/coreboot/kgpe-d16-bmc-port-offer.php
+in the same year. The initial source code is still available at
+[Raptor Engineering's site](https://www.raptorengineering.com/coreboot/kgpe-d16-bmc-port-status.php).
+
+The platform support lived on the main coreboot branch for 3 years until the
+[4.12 release](https://doc.coreboot.org/releases/coreboot-4.12-relnotes.html)
+where features like C bootblock, postcar phase and relocatable ramstage support
+became mandatory. The board support has been dropped from the main branch due
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: rephrased to `board support could not be kept on the main branch`

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+current performance per dollar expectations, but there is always some tradeoff.
+What is best about these processors is that 6100 and 6200 series Opteron CPUs
+run very well without microcode, there is no PSP and the graphics
+initialization is done without VBIOS option ROM. Everything in the firmware may
+be handled by pure open-source code.
+
+The open-source BIOS for ASUS KGPE-D16 has been implemented by Timothy Pearson
+from Raptor Engineering in 2017 by crowdfunding:
+https://review.coreboot.org/q/topic:raptor-asus-kgpe-d16 
+The same was done for OpenBMC
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Done

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+impressive, but also energy-hungry as well which may not be acceptable for
+current performance per dollar expectations, but there is always some tradeoff.
+What is best about these processors is that 6100 and 6200 series Opteron CPUs
+run very well without microcode, there is no PSP and the graphics
+initialization is done without VBIOS option ROM. Everything in the firmware may
+be handled by pure open-source code.
+
+The open-source BIOS for ASUS KGPE-D16 has been implemented by Timothy Pearson
+from Raptor Engineering in 2017 by crowdfunding:
+https://review.coreboot.org/q/topic:raptor-asus-kgpe-d16
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Done

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+security by obscurity, falling deeper into the pit they created themselves,
+bound by laws that were supposed to protect them. As a result open-source
+firmware community has to struggle and push vendors into openness or to provide
+means to run open firmware on their products. The openness and possibilities to
+run open firmware is gradually decreasing over time as vendors create more and
+more binary blobs, offload various operation to another entities (e.g. AMD PSP
+or Intel ME). These entities are often fed with more firmware and blobs, often
+closed and proprietary with source code being the vendor's restricted secret.
+In the light of this threat we turn our eyes to older platforms that were free
+from firmware blobs, embedded secondary microcontrollers in chipsets and were
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Added

File: blog/content/post/2022-01-17-kgpe_D16_status.md (hunk 1 above)
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Added

File: blog/content/post/2022-01-17-kgpe_D16_status.md (hunk 2 above)
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Added

File: blog/content/post/2022-01-17-kgpe_D16_status.md (hunk 3 above)
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Added

File: blog/content/post/2022-01-17-kgpe_D16_status.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Will that suffice?

[the remark this answers is not in the document]
Comment (pietrushnic) [reply blog/content/post/2022-01-17-kgpe_D16_status.md]: Yes.

File: blog/content/post/2022-01-17-kgpe_D16_status.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+Be sure to attend this years' FOSDEM'22 where I will be talking about current
+[status of AMD platforms](https://fosdem.org/2022/schedule/event/osf_on_amd_3rd/)
+(including KGPE-D16) in open source firmware in the Open Source Firmware, BMC
+and bootloader devroom.
+
+Also feel invited for the [FOSDEM'22 after-party](https://3mdeb.com/events/#fosdem-22)
+organized right after the devroom track where we will be discussing various
+aspects of related to firmware and bootloaders. 3mdeb is also organizing a
+[vPub 0x4](https://3mdeb.com/events/#vpub-0x4) on the 17th of February, don't
```
Comment (pietrushnic) [inline blog/content/post/2022-01-17-kgpe_D16_status.md:150]:
```suggestion
[vPub](https://vpub.dasharo.com) on the 17th of February, don't
```

**Pull request merged**